### ISSUE TITLE
Course?

### ISSUE THREAD

**Issue by Manouchehri** [angr/angr-doc#70]
As @zardus mentioned in #66, it might be fun to write an angr class/course that helps new users progress from basic to complex reverse/crackmes. It would be similar to `examples.md`, but with the binaries and scripts explained in more detail.

My question is, where do you guys want to put this? In angr-doc? In a new repo?

**Comment by ltfish** [comment]
`angr-doc` should be good. As I see it, we should have it as the first part of the online documentation (http://docs.angr.io).

**Comment by zardus** [comment]
Yep, agreed. COURSE.md would be slick :-)

**Comment by giovannivigna** [comment]
Having a class available with video and similar things would be fantastic.

G

**Comment by kirschju** [comment]
Heh, I already have some material ready from my teaching activities + some talks. Would you prefer slides or writeups?

**Comment by zardus** [comment]
In the ideal world, we'd have the following (IMO):

- a COURSE.md that walks people through getting started with angr, with examples of increasing difficulty that require the use of different angr functions
- slides that explain the ideas behind these angr functions (linked from the sections of COURSE.md)
- where possible, linked videos explaining the concept or presenting the slides (in a youtube tutorial style, I guess)

Then people could go through COURSE.md, getting familiar with different angr capabilities and jumping off to the slides and videos when they need further explanation.

So, in summary, probably everything would be useful. If someone wants to create a starting template for COURSE.md and pile it up with some stuff, I think that'd be the way to go :-)

**Comment by zardus** [comment]
So I made the stub that we can expand when there's time: https://github.com/angr/angr-doc/blob/master/docs/course.md

**Comment by P1kachu** [comment]
I started [something](https://github.com/angr/angr-doc/pull/74) if any of you want to take a look. I'm not sure about the format though.

**Comment by R3x** [comment]
Well, I am new to angr and I would definitely love some sort of course which introduced people to all the features of the cool engine you guys have developed. Many of the scripts in the examples doc are not clearly documented. So do kindly go forward with this thing that you have planned.

**Comment by kirschju** [comment]
Short update on this matter: We can provide another angr mini course that was being developed during a seminar course over here. You can find the contents [here](https://kirschju.re/static/angr-mini-course.tar.gz). Please let us know whether you'd like to receive this as a pull request.

**Comment by Manouchehri** [comment]
@kirschju Looks good! Did you mean to exclude `src/` though?

**Comment by zardus** [comment]
That's pretty awesome, @kirschju! I think it'd make a great PR.

**Comment by PeterGreth** [comment]
Okay great! I will create a pull request soon.
@zardus Do you want to keep all binaries in one directory? Is the following order of the courses okay?
1. Stashes
2. Path explosion, hooking
3. Symbolic input, command line parameters, constraints
4. CFGs
5. Symbolic Execution Example, SimStates

**Comment by zardus** [comment]
Sounds great. For the course, I think it's ok to spread out (and even duplicate) the binaries per module.

**Comment by Manouchehri** [comment]
I think this is already covered by the docs now.
